pennylane/fourier/utils.py

```diff
@@ -42,3 +42,73 @@ def to_dict(coeffs):
     nvecs = product(*n_ranges)
 
     return {nvec: coeffs[nvec] for nvec in nvecs}
+
+
+def _get_spectrum(op, decimals):
+    r"""Extract the frequencies contributed by an input-encoding gate to the
+    overall Fourier representation of a quantum circuit.
+
+    If :math:`G` is the generator of the input-encoding gate :math:`\exp(-i x G)`,
+    the frequencies are the differences between any two of :math:`G`'s eigenvalues.
+    We only compute non-negative frequencies in this subroutine.
+
+    Args:
+        op (~pennylane.operation.Operation): :class:`~.pennylane.Operation` to extract
+            the frequencies for
+        decimals (int): Number of decimal places to round the frequencies to
+
+    Returns:
+        set[float]: non-negative frequencies contributed by this input-encoding gate
+    """
+    no_generator = False
+    if hasattr(op, "generator"):
+        g, coeff = op.generator
+
+        if isinstance(g, np.ndarray):
+            matrix = g
+        elif hasattr(g, "matrix"):
+            matrix = g.matrix
+        else:
+            no_generator = True
+    else:
+        no_generator = True
+
+    if no_generator:
+        raise ValueError(f"Generator of operation {op} is not defined.")
+
+    matrix = coeff * matrix
+    # todo: use qml.math.linalg once it is tested properly
+    evals = np.linalg.eigvalsh(matrix)
```

Comment:
This was using `qml.math.real(np.linalg.eigvals(matrix))` before.
As we know the `matrix` to be Hermitian, we may use `eigvalsh` instead, with the advantage of obtaining *sorted, real* values right away.

Reply:
This is inherited from the original `spectrum`, see the commented line 80. I am not sure, whether `qml.math.linalg` is "tested properly" by now? Happy to make this diffable!

Reply:
Ah, actually no need to make the change here - I just realized the _original_ function was non-differentiable, so that would be out of scope for this PR.

However, definitely something worth considering for a future PR :)